pkg/process/event_stream.go

```diff
@@ -0,0 +1,168 @@
+package process
+
+import (
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	gardenerazurev1alpha1 "github.com/gardener/gardener-extension-provider-azure/pkg/apis/azure/v1alpha1"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/kyma-incubator/metris/pkg/edp"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	nodeInstanceTypeLabel = "node.kubernetes.io/instance-type"
+	// storageRoundingFactor rounds of storage to 32. E.g. 17 -> 32, 33 -> 64
+	storageRoundingFactor = 32
+
+	Azure = "azure"
+)
+
+type EventStream struct {
+	Metric     edp.ConsumptionMetrics
+	KubeConfig string
+}
+
+type Input struct {
+	shoot    *gardencorev1beta1.Shoot
+	nodeList *corev1.NodeList
+	pvcList  *corev1.PersistentVolumeClaimList
+	svcList  *corev1.ServiceList
+}
+
+type NodeInfo struct {
+	cpu    int
+	memory int
+}
+
+func (inp Input) Parse(providers *Providers) (*edp.ConsumptionMetrics, error) {
+
+	if inp.nodeList == nil {
+		return nil, fmt.Errorf("no nodes data to compute metrics on")
+	}
+	if inp.shoot == nil {
+		return nil, fmt.Errorf("no shoot data to compute metrics on")
+	}
+
+	metric := new(edp.ConsumptionMetrics)
+	provisionedCPUs := 0
+	provisionedMemory := 0.0
+	providerType := inp.shoot.Spec.Provider.Type
+	vmTypes := make(map[string]int)
+
+	nodeStorage := int64(0)
+	pvcStorage := int64(0)
+	volumeCount := 0
+	vnets := 0
+
+	for _, node := range inp.nodeList.Items {
+		nodeType := node.Labels[nodeInstanceTypeLabel]
+		nodeType = strings.ToLower(nodeType)
+
+		// Calculate CPU and Memory
+		vmFeatures := providers.GetFeatures(providerType, nodeType)
+		if vmFeatures == nil {
+			return nil, fmt.Errorf("providerType: %s and nodeType: %s does not exist in the map", providerType, nodeType)
+		}
+		provisionedCPUs += vmFeatures.CpuCores
+		provisionedMemory += vmFeatures.Memory
+		vmTypes[nodeType] += 1
+
+		// Calculate node storage
+		nodeStorage += int64(vmFeatures.Storage)
```

Comment:
why `int64` if the storage in the struct is int?

Reply:
Changed to int64